canonical/vanilla-framework
# Trap focus within the sidenav on small screens [WD-2277]

## Done

Added an example of how the focus should be controlled within a sidenav drawer to the already existing JS code of the sidenav toggle script.

Fixes https://github.com/canonical/vanilla-framework/issues/4560

## QA

- Open [a sidenav example](https://vanilla-framework-4680.demos.haus/docs/examples/patterns/side-navigation/expandable).
- Resize the window so that its width becomes lower than `1036px`. The sidenav switches from "pinned" to "drawer" mode.
- Hit the `Tab` key multiple times to check that the focus is trapped within the sidenav.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
